### PR TITLE
feat: added support for 'file://' protocol in demand spec and fixed e…

### PR DESCRIPTION
**Pull request by grisha87**
…xamples

### CODE REVIEW

File: src/golem-network.ts [hunk 1]
```diff
@@ -100,6 +104,8 @@ export class GolemNetwork {
         basePath: this.options.api?.url,
       });
 
+      this.gftpStorageProvider = new GftpStorageProvider();
```
Comment (mgordel) [inline src/golem-network.ts:107]: this should be dependent on `dataTransferProtocol` option

File: examples/market/scan.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-  payment: {
+(async () => {
+  const glm = new GolemNetwork({
+    logger: pinoPrettyLogger({
+      level: "debug",
+    }),
     payment: {
-      network: "holesky",
-      driver: "erc20",
+      payment: {
```
Comment (mgordel) [inline examples/market/scan.ts:14]: do we have to double nest it?

**Review by mgordel — APPROVED** [review]

File: examples/market/scan.ts (hunk 2 above)
Comment (grisha87) [reply examples/market/scan.ts:14]: This became required after recent merges in the branch. We will remove the double nesting for sure. Probably even next week.

File: src/golem-network.ts (hunk 1 above)
Comment (grisha87) [reply src/golem-network.ts:107]: I've applied this remark. The only issue is that this capability is not really implemented in the WS-based GFTP transport:

```
Failed to run example on Golem GolemInternalError: Not implemented
    at WebSocketBrowserStorageProvider.publishFile (/home/ggodlewski/proj/golem/golem-js/dist/shared-ChCc8mz0.mjs:719:15)
    at GftpServerAdapter.publishFile (/home/ggodlewski/proj/golem/golem-js/dist/shared-ChCc8mz0.mjs:5963:44)
    at MarketModuleImpl.applyLocalGVMIServeSupport (/home/ggodlewski/proj/golem/golem-js/dist/shared-ChCc8mz0.mjs:3528:139)
    at MarketModuleImpl.buildDemand (/home/ggodlewski/proj/golem/golem-js/dist/shared-ChCc8mz0.mjs:3501:39)
    at <anonymous> (/home/ggodlewski/proj/golem/golem-js/examples/local-image/serveLocalGvmi.ts:48:50)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  previous: undefined
}
```

**Pull request merged**